test/sanitize_json.d

```diff
@@ -39,25 +39,50 @@ int main(string[] args)
     }
 
     auto json = parseJSON(readText(inFilename));
-    sanitize(json.array);
+    sanitize(json);
 
-    outFile.write(json.toJSON(true));
+    outFile.write(json.toJSON(true, JSONOptions.doNotEscapeSlashes));
     return 0;
 }
 
-void sanitize(JSONValue[] rootArray)
+void sanitize(JSONValue root)
 {
-    foreach (ref obj; rootArray)
+    if (root.type == JSON_TYPE.ARRAY)
     {
-        auto kind = obj.object["kind"].str;
-        if (kind == "compilerInfo")
-            sanitizeCompilerInfo(obj.object);
-        else if (kind == "buildInfo")
-            sanitizeBuildInfo(obj.object);
-        else if(kind == "module")
-            sanitizeSyntaxNode(obj);
-        else if(kind == "semantics")
-            sanitizeSemantics(obj.object);
+        sanitizeSyntaxNode(root);
+    }
+    else
+    {
+        assert(root.type == JSON_TYPE.OBJECT);
+        auto rootObject = root.object;
+        {
+            auto compilerInfo = rootObject.get("compilerInfo", JSONValue.init);
+            if (compilerInfo.type != JSON_TYPE.NULL)
+            {
+                sanitizeCompilerInfo(compilerInfo.object);
+            }
+        }
+        {
+            auto buildInfo = rootObject.get("buildInfo", JSONValue.init);
+            if (buildInfo.type != JSON_TYPE.NULL)
+            {
+                sanitizeBuildInfo(buildInfo.object);
+            }
+        }
+        {
+            auto modules = rootObject.get("modules", JSONValue.init);
+            if (modules.type != JSON_TYPE.NULL)
+            {
+                sanitizeSyntaxNode(modules);
+            }
+        }
+        {
+            auto semantics = rootObject.get("semantics", JSONValue.init);
+            if (semantics.type != JSON_TYPE.NULL)
+            {
+                sanitizeSemantics(semantics.object);
+            }
+        }
```

Comment:
DRY idea: 

```d
static foreach (e; [tuple("compilerInfo", "sanitizeCompilerInfo")
                    tuple("buildInfo", "sanitizeBuildInfo")])])
{{
    auto infonfo = rootObject.get(e[0], JSONValue.init);
    if (compilerInfo.type != JSON_TYPE.NULL)
        mixin(e[1] ~ "(info.object);");
}}
```

It's okay to use newer compiler features here as DMD + DRuntime + Phobos are just freshly built from master.